RE: [Intent] Count my open tabs 1267

xlisachan opened the pull request:
Based on the intents mentioned in #1267, I think it also makes sense to provide the number of tabs on the current window only, as opposed to the total open tabs of all windows. Please see a screenshot of the change below.

<img width="368" alt="Screen Shot 2020-03-13 at 4 35 48 PM" src="https://user-images.githubusercontent.com/28129806/76658369-5d912500-654a-11ea-84db-c6379e13b063.png">

I noticed that after making a query like "Who created Breaking Bad?", it counts the hidden pinned search tab (indicated on line 42 of search.toml), which may confuse users. The Telemetry ping in the console also counts the hidden tab as well. Can you please review and provide guidance when you have a moment?

<img width="670" alt="Screen Shot 2020-03-13 at 5 05 10 PM" src="https://user-images.githubusercontent.com/28129806/76659432-d2655e80-654c-11ea-8512-dac91b24804b.png">

<img width="394" alt="Screen Shot 2020-03-13 at 5 03 54 PM" src="https://user-images.githubusercontent.com/28129806/76659412-c083bb80-654c-11ea-9372-1efb88abc70b.png">

ianb commented:
With regard to hidden tabs, I think you are right that both here and in telemetry.js we should exclude hidden tabs from the count. It looks like `.query({hidden: false})` will do that.

ianb commented on extension/intents/tabs/tabs.js:
```diff
@@ -556,3 +556,23 @@ if (!buildSettings.android) {
+      answer: {
+        num: `${tabs.length}`,
+        text: "Open tabs",
+        eduText: "Click mic and say 'gather all Google tabs'",
```
It looks like we actually don't support "gather", though that could be added to the "collect" phrases that we have (`(collect | gather)`).

I noticed @awallin put single quotes around `'gather all Google tabs'` in the mockup, but generally we've been using double quotes (`"gather all Google tabs"`).

Also could you rename `num` to be `largeText`? It doesn't have to be a number, but it will be very large! I can imagine `mediumText` being added soon as well for some other information.

xlisachan replied:
I make updates to support "gather", use double quotes, and rename `num`.

ianb commented on extension/intents/tabs/tabs.js:
```diff
@@ -556,3 +556,23 @@ if (!buildSettings.android) {
+      },
+    };
+    await browser.runtime.sendMessage({
+      type: "showSearchResults",
```
Hmm... we should probably make a followup to rename this, I can see why it would be the easy thing to work with here, but "showSearchResults" also isn't the right name anymore.

xlisachan replied:
I was actually in the middle of creating a new one for (what is now) AnswerCard, but I saw a note saying this issue should be a straightforward update, so I thought I was going deep down a rabbithole. If it's decided that it will be renamed, I would like to give it another shot!

ianb commented on extension/popup/popupView.jsx:
```diff
@@ -589,23 +595,50 @@ const SearchResultsContent = ({
     onNextSearchResultClick();
   };
 
+  const searchCard = () => (
```
We've been capitalizing these functions that return markup, so SearchCard would be a better name.

xlisachan replied:
I will make this update!

ianb commented on extension/popup/popupView.jsx:
```diff
@@ -589,23 +595,50 @@ const SearchResultsContent = ({
+    </button>
+  );
+
+  const qaCard = () => (
```
I don't understand the name `qaCard`? Maybe `AnswerCard` would be the better name?

xlisachan replied:
It was referred to as the Q&A card in issue #1267 so I named it `qaCard`, haha. `AnswerCard` is definitely a better name, I will make this change.

xlisachan commented:
@ianb Thanks for all the feedback! I have made the following updates:
1. Excluded hidden tabs in the counter in tabs.js
2. Excluded hidden tabs in the numberOfTabs counter in telmetry
3. Added support for "gather" in tabs.toml
4. Updated single quotes to double quotes in IntentFeedback tag
5. Changed variable `num` to `largeText` and updated the className as well
6. Capitalized markup function `SearchCard()` 
7. Renamed `qaCard` to `AnswerCard()`

Please review and let me know if you have any questions.

ianb commented:
Looking really good, thanks! And you got some extra improvements in there along the way...
